### DISCUSSION TITLE
OSQuery Agent Not Linking into Fleet

### DISCUSSION THREAD

**Discussion by ow111** [Security-Onion-Solutions/securityonion#4022] [Unsupported Versions] [answered]
Distributed Deployment running SO v2.3.40 
Airgapped Network 

Hi there, Have been deploying OSQuery across a network and been seeing some strange issues with linking back into the Manager into Fleet. For 95% of hosts they instantly populate and work flawlessly - however there are a few Hosts not taking it.
Mix of OS - Windows/Ubuntu/CentOS seen across all..
Tests carried out - 1. DNS lookup of the manager 2. sudo salt-call pillar.get global (url_base showing FQDN and fleet_hostname showing just hostname) 3. removal of package and deleted all metadata - then tried a reinstall 4. Tested port 8090 on the manager from hosts 5. editing launcher.flags to use FQDN/IP address  All of these troubleshooting steps have not yielded any luck.

Some logs seen in Application eventlog:
Cannot activate kolide_grpc config plugin: Extension socket not available 
More can be provided

Also a side note - is it possible to Run Fleet on a forward node?

**Comment by defensivedepth** [comment, marked as answer]
Best thing to do is check out the Launcher/Osqueryd logs on a few of the hosts that are not enrolling (restart the service on them so you can easily see it trying to connect):

- Windows --> checkout the Launcher logs under the Application event channel
- Linux --> `sudo systemctl status launcher.so-launcher.service -l`

**Reply by ow111** [reply]
Ok - I'll get back to you next week with some logs

**Reply by ow111** [reply]
Here are some logs from one of the Windows Hosts we are having issues with, In order from service being started:
msg="E0504 5624 shutdown.cpp:69 , Cannot activate kolide_grpc config plugin: Extension socket not available: \\\\.\\pipe\\kolide-osquery-XXXXXXXXXXXXXXX.22915 caller = shutdown.cpp:69     (XXX = Specific to different host as expected

msg="W0504 5624 init.cpp:587 , Error reading config: Extension socket not available: \\\\.\\pipe\\kolide-osquery-XXXXXXXXX.22915" caller = init.cpp:587

Next log is a connection to https://notary.kolide.co/_notary_server/health\ - Failed since air gapped system.

msg="E0504 5624 init.cpp:69, Cannot Activate kolide_grpc logger plugin: Extension socket not available: \\\\.\\pipe\\kolide-osquery-XXXXXXXXX.22915" caller = init.cpp:69

msg="E0504 5624 init.cpp:69 , Cannot activate kolide_grpc distributed plugin: Extension socket not available: \\\\.\\pipe\\kolide-osquery-XXXXXXXXX.22915" caller = init.cpp:69

msg="I0504 5624 init.cpp:867, Event publisher not enabled: ntfs_event_publisher: NTFS event publisher disabled via configuration caller=events.cpp:867 

path="C:\\Program Files\\Kolide\\Launcher-so-launcher\\bin\\osqueryd.exe" sizeBytes=14811648 mode= -rw -rw -rw- sha256=xxxx msg="Error running osquery command" err= "exit status 78"

caller=launcher.go:122 err="enrolling host: query enrollment details, (even with retries): done trying: query enrollment details: could not query the extension manager client: the pipe is being closed"

**Reply by defensivedepth** [reply]
hmmmmm.

When you get a chance, can you run the following on the Manager (it will refresh the osquery packages - will take up to 5 min, though the command returns quickly):

Then download a fresh MSI, uninstall the current MSI and install the new one?

**Reply by ow111** [reply]
Did you miss the command out? Dont see it

**Reply by ow111** [reply]
Here are some logs from CentOS machine (a Search node in our cluster which is showing offline/MIA in fleet)

launcher[28898]: {"caller":"query_target_updater.go:21","msg":"query target updater started","severity":"info","ts":"2021-05-04T11:05:21.60669816Z"}
launcher[28898]: {"caller":"updater.go:74","msg":"updater started","severity":"info","ts":"2021-05-04T11:05:21.607221125Z"}
launcher[28898]: {"arg0":"/usr/local/so-launcher/bin/osqueryd","args":"/usr/local/so-launcher/bin/osqueryd --logger_plugin=kolide_grpc --distributed_plugin=kolide_grpc --disable_distributed=false --distributed_interval=5 --pack_delimiter=: --host_identifier=uuid --force=true --disable_watchdog --utc --config_refresh=300 --config_accelerated_refresh=30 --pidfile=/var/so-launcher/securityonion/osquery.pid --database_path=/var/so-launcher/securityonion/osquery.db --extensions_socket=/var/so-launcher/securityonion/osquery.sock --extensions_autoload=/var/so-launcher/securityonion/osquery.autoload --extensions_timeout=10 --config_plugin=kolide_grpc","caller":"runtime.go:566","msg":"launching osqueryd","severity":"info","ts":"2021-05-04T11:05:21.636592392Z"}
osqueryd[28908]: osqueryd started [version=4.5.1]
launcher[28898]: {"caller":"","component":"osquery","level":"stderr","msg":"[/usr/local/so-launcher/bin/osquery-extension.ext --socket /var/so-launcher/securityonion/osquery.sock --timeout 10 --interval 3]","severity":"info","ts":"2021-05-04T11:05:21.698292078Z"}
launcher[28898]: {"caller":"extension.go:128","msg":"extension started","severity":"info","ts":"2021-05-04T11:05:22.040854451Z"}
launcher[28898]: {"caller":"interface.cpp:110","component":"osquery","level":"stderr","msg":"I0504 11:05:22.048300 28929 interface.cpp:110] Registering extension (kolide, 18828, version=, sdk=)","severity":"info","ts":"2021-05-04T11:05:22.04840775Z"}
launcher[28898]: {"caller":"options.cpp:92","component":"osquery","level":"stderr","msg":"W0504 11:05:22.085188 28908 options.cpp:92] Cannot set unknown or invalid flag: enable_windows_events_publisher\nW0504 11:05:22.085286 28908 options.cpp:92] Cannot set unknown or invalid flag: enable_windows_events_subscriber","severity":"info","ts":"2021-05-04T11:05:22.085353295Z"}
launcher[28898]: {"caller":"events.cpp:867","component":"osquery","level":"stderr","msg":"I0504 11:05:22.090373 28908 events.cpp:867] Event publisher not enabled: auditeventpublisher: Publisher disabled via configuration","severity":"info","ts":"2021-05-04T11:05:22.090503708Z"}
launcher[28898]: {"caller":"events.cpp:867","component":"osquery","level":"stderr","msg":"I0504 11:05:22.090468 28908 events.cpp:867] Event publisher not enabled: syslog: Publisher disabled via configuration","severity":"info","ts":"2021-05-04T11:05:22.090577773Z"}


Some CentOS machines/ Ubuntu machines take the package straight away and link into Fleet - Same with 95% of the Windows estate. For some extra info , the only windows machines that are playing up are hosts that host fileshares..

**Reply by ow111** [reply]
I've managed to refresh the packages - uninstall from Windows machine and rebooted. Removed the old metadata and reinstalled the osquery package, Still no joy.. :(

**Reply by defensivedepth** [reply]
@ow111 Is this still an issue?

**Reply by ow111** [reply]
Hey, after a complete flatten of all Windows Servers from scratch , fleet now seems to be working properly. 
As a side note: Is there a supported way to switch windows domains that SO is configured for? Or is a full redeployment the best option?

**Reply by defensivedepth** [reply]
@ow111 Can you clarify what you mean by switch windows domains?

**Reply by ow111** [reply]
Sorry, let me get myself straight here. I need to switch the IP addresses for both search nodes and a manager node, to a completely new subnet. Both of the searchnodes call back into the manager ok after changing the minion.sls files and the global file, but for some reason the cross cluster searching and elasticsearch element of the search nodes seems to be malfunctioning. so-status shows them as up and running yet port 9200&9300 are unresponsive. I have made sure that the new searchnode ips are part of the searchnodes and minion hostgroups in so-firewall, same goes for the manager. Is this just a full redeployment? It was working completely fine before migrating the networking side. Hostnames have remained the same, just IP addresses have been changed

**Reply by defensivedepth** [reply]
`so-ip-update` is experimental and only supports standalone, not distributed --> https://docs.securityonion.net/en/2.3/ip.html
